### ISSUE TITLE
Semantic code highlighting via Language Server Protocol

### ISSUE THREAD

**Issue by mosra** [mosra/m.css#109]
Pygments is great and all, but it can only do so much to highlight the code -- and especially in typeless languages (Python) or *insanely hard to parse* languages (C++) it can't provide type links, "go to definition" or anything like that. For C++ there's Clang-based and Pygments-compatible https://github.com/Oberon00/synth, but it seems abandoned.

I think [Language Server Protocol](https://github.com/Microsoft/language-server-protocol) could help in this area -- it could be enough for m.css to pretend it's an editor and ask the server what each and every token is.

Existing implementations (among others, there's Rust and Go):

- for C++: https://clang.llvm.org/extra/clangd/
- for Python: https://github.com/palantir/python-language-server
- for Latex: https://texlab.netlify.com/ :scream:

**Comment by marzer** [comment]
CC @marzer (just tagging myself here for future reference)
